anvill/include/anvill/Lifters/EntityLifter.h

```diff
@@ -54,7 +55,8 @@ class EntityLifter {
   explicit EntityLifter(
       const LifterOptions &options,
       const std::shared_ptr<MemoryProvider> &mem_provider_,
-      const std::shared_ptr<::anvill::TypeProvider> &type_provider_);
+      const std::shared_ptr<::anvill::TypeProvider> &type_provider_,
+      const Program &program);
```

Comment:
Pass through the control-flow provider instead; we eventually want to get rid of or factor out `Program`.

Reply:
Perhaps instead we should put the providers into lifter options? What do you think? I.e. one needs to do `LifterOptions::Create(providers here)` to get back a value type, then you can fiddle with the options. Another possibility would be a lifter context which has a lifter options and each provider.